model/src/main/kotlin/Provenance.kt

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2017-2018 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package com.here.ort.model
+
+import com.fasterxml.jackson.annotation.JsonInclude
+
+import java.time.Instant
+
+/**
+ * Provenance information about the scanned source code.
```

Comment:
Similar to like we do in `DownloadResult`, we should probably says that "Either [sourceArtifact] or [vcsInfo] is set to a non-null value".

Reply:
Done. Also added a check that not both can be set to a non-null value.